packages/gitbook/src/app/(site)/fetch.ts

```diff
@@ -33,19 +33,28 @@ export async function fetchContentData() {
                 organizationId: content.organizationId,
                 siteId: content.siteId,
                 siteShareKey: content.siteShareKey,
+                siteSpaceId: content.siteSpaceId,
             }),
         ]);
 
     const site = siteStructure.site;
-    const spaces = siteStructure.spaces;
+
+    const sections = siteStructure.sections;
+    const section = siteStructure.sections?.find((section) => section.id === content.siteSectionId);
```

Comment:
If it's a sections site then `section` should ALWAYS be defined else we try. I think you can put a condition that 
```js
if(sections) {
   assertIsDefined(section);
}
```